src/lib.rs

```diff
@@ -366,17 +366,20 @@ to write correct code for Unix, at the cost of getting a corner case wrong on
 Windows.
 */
 
-#![cfg_attr(not(feature = "std"), no_std)]
+#![no_std]
 #![allow(dead_code)]
 
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
 #[cfg(feature = "std")]
-extern crate core;
+extern crate std;
```

Comment:
If you'd prefer, I can make this `#[cfg(any(test, feature = "std"))]` and remove the `alloc` gates from tests.

Reply:
Yes, I would like that. Otherwise, I think writing tests becomes too annoying. With that said, I do think that things like `cargo t --no-default-features` and `cargo t --no-default-features --features alloc` should at least compile fine (and run any relevant tests, if any, probably none). So I think most tests do need to be gated on `#[cfg(feature = "std")]`.